openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+    name : str, optional
+        Name of the tally. If not specified, the name is the empty string.
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the tally
+    name : str
+        Name of the tally
+    filters : list of openmc.Filter
+        List of specified filters for the tally
+    nuclides : list of openmc.Nuclide
+        List of nuclides to score results for
+    scores : list of str
+        List of defined scores, e.g. 'flux', 'fission', etc.
+    estimator : {'analog', 'tracklength', 'collision'}
+        Type of estimator for the tally
+    triggers : list of openmc.Trigger
+        List of tally triggers
+    num_scores : int
+        Total number of scores
+    num_filter_bins : int
+        Total number of filter bins accounting for all filters
+    num_bins : int
+        Total number of bins for the tally
+    shape : 3-tuple of int
+        The shape of the tally data array ordered as the number of filter bins,
+        nuclide bins and score bins
+    filter_strides : list of int
+        Stride in memory for each filter
+    num_realizations : int
+        Total number of realizations
+    with_summary : bool
+        Whether or not a Summary has been linked
+    sum : numpy.ndarray
+        An array containing the sum of each independent realization for each bin
+    sum_sq : numpy.ndarray
+        An array containing the sum of each independent realization squared for
+        each bin
+    mean : numpy.ndarray
+        An array containing the sample mean for each bin
+    std_dev : numpy.ndarray
+        An array containing the sample standard deviation for each bin
+    derived : bool
+        Whether or not the tally is derived from one or more other tallies
+    sparse : bool
+        Whether or not the tally uses SciPy's LIL sparse matrix format for
+        compressed data storage
+    derivative : openmc.TallyDerivative
+        A material perturbation derivative to apply to all scores in the tally.
+
+    """
+
+    next_id = 1
+    used_ids = set()
+
+    def __init__(self, tally_id=None, name=''):
+        # Initialize Tally class attributes
+        self.id = tally_id
+        self.name = name
+        self._filters = cv.CheckedList(_FILTER_CLASSES, 'tally filters')
+        self._nuclides = cv.CheckedList(_NUCLIDE_CLASSES, 'tally nuclides')
+        self._scores = cv.CheckedList(_SCORE_CLASSES, 'tally scores')
+        self._estimator = None
+        self._triggers = cv.CheckedList(openmc.Trigger, 'tally triggers')
+        self._derivative = None
+
+        self._num_realizations = 0
+        self._with_summary = False
+
+        self._sum = None
+        self._sum_sq = None
+        self._mean = None
+        self._std_dev = None
+        self._with_batch_statistics = False
+        self._derived = False
+        self._sparse = False
+
+        self._sp_filename = None
+        self._results_read = False
+
+    def __repr__(self):
+        string = 'Tally\n'
```

Comment:
although not a keyword in python, we shouldn't use `string` since it's a type. It would be like naming a variable `int`.
We should call it something along the lines of `reprString`